frontend/src/host_orchestrator/orchestrator/userartifacts.go

```diff
@@ -69,20 +80,35 @@ type UserArtifactsManagerOpts struct {
 // An implementation of the UserArtifactsManager interface.
 type UserArtifactsManagerImpl struct {
 	UserArtifactsManagerOpts
+	// The root directory where to store artifacts with legacy API calls.
+	legacyDir   string
+	workDir     string
+	tempDir     string
+	sharedDir   string
+	mutexes     sync.Map
+	chunkStates sync.Map
 }
 
 // Creates a new instance of UserArtifactsManagerImpl.
 func NewUserArtifactsManagerImpl(opts UserArtifactsManagerOpts) *UserArtifactsManagerImpl {
+	sharedDir := filepath.Join(opts.RootDir, "shared")
 	return &UserArtifactsManagerImpl{
 		UserArtifactsManagerOpts: opts,
+		legacyDir:                filepath.Join(opts.RootDir, "legacy"),
+		workDir:                  filepath.Join(opts.RootDir, "working"),
```

Comment:
clean up workDir once process is terminated or at started. because it's dummy data once the context is lost.